src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+            1: "region_series",
+            2: "case_series",
+            3: "report_year_series",
+        }
+    )
+    variable_fields = fields.variable_fields.str.split(" : ", expand=True).rename(
+        columns={0: "topic", 1: "subtopic", 2: "variable_name", 3: "dimension"}
+    )
+    return fields.join(variable_fields)
+
+
+def get_category_info(category_name: pd.Series) -> pd.Series:
+    """Break human-readable category name into machine-readable fields.
+
+    Fortunately the only field we're pulling out of the category so far is the
+    region, which is the last of two comma-separated fields.
+    """
+    fields = category_name.str.rsplit(",", n=1).str.get(-1).rename("region_category")
+    return fields
+
+
+def filter_enrich_sanitize(
+    raw_df: pd.DataFrame, relevant_series_names: tuple[str]
+) -> pd.DataFrame:
+    """Basic cleaning steps common to all AEO tables.
+
+    1. Filter the AEO rows based on the series name
+    2. Break the series name and category names into useful fields
+    3. Sanitize strings & turn data values into a numeric field
+    4. Make some defensive checks about data from multiple sources that
+       *should* agree.
+    """
+    filtered = raw_df.loc[raw_df.series_name.str.startswith(relevant_series_names)]
+
+    enriched = pd.concat(
+        [
+            filtered,
+            get_series_info(filtered.series_name),
+            get_category_info(filtered.category_name),
+        ],
+        axis="columns",
+    ).drop(
+        columns=[
+            "series_name",
+            "category_name",
+            "variable_fields",
+        ]
+    )
+
+    sanitized = enriched.assign(
+        **{
+            colname: __sanitize_string(enriched[colname])
+            for colname in enriched.columns
+            if colname not in {"projection_year", "value"}
+        }
+    ).pipe(
+        lambda df: df.assign(
+            report_year_series=df.report_year_series.str.replace("aeo", ""),
+            value=pd.to_numeric(df.value.str.replace("- -", "")),
+        )
+    )
+
+    assert (
+        sanitized.region_series == sanitized.region_category
+    ).all(), "Series and category must agree on region!"
+    assert (
+        sanitized.case_series == sanitized.model_case_eiaaeo
+    ).all(), "Series and taxonomy must agree on case!"
+
+    # TODO 2024-04-20: one day we'll include report year in the extract step -
+    # at that point we'll have to compare that with the report year extracted
+    # from the series name.
+    return sanitized.drop(columns=["region_series", "case_series"]).rename(
+        columns={"region_category": "region", "report_year_series": "report_year"}
+    )
+
+
+def unstack(df: pd.DataFrame, eventual_pk: list[str]):
+    """Unstack the values by the various variable names provided.
+
+    2024-04-27: NB it's important that `projection_year` is the last level of
+    index, so that when we sort and diff() we are actually subtracting year
+    over year.
```

Comment:
what is NB? also should this be converted into a `Raises:` section since you added the assert in here?

Reply:
["Nota bene," Latin for "note well"](https://en.wikipedia.org/wiki/Nota_bene) - IDK why I chose to use this instead of the more normal word "Note," maybe I was spending too much time reading legalese in leases.

Reply:
lolol 📜